improver_tests/utilities/test_CalculateClimateAnomalies.py

```diff
@@ -0,0 +1,314 @@
+    result = plugin.process(diagnostic_cube, mean_cube)
+    np.testing.assert_allclose(result.data, expected_anomalies, rtol=1e-5)
+    assert result.name() == diagnostic_cube.name() + "_anomaly"
+    assert result.units == "K"
+    assert "reference_epoch" in [coord.name() for coord in result.coords()]
+    assert (
+        "(CellMethod(method='anomaly', coord_names=('reference_epoch',), intervals=(), comments=())"
+        in str(result.cell_methods)
+    )
+
+
+@pytest.mark.parametrize(
+    "fixture_name", ["diagnostic_cube", "diagnostic_cube_multiple_time_points"]
+)
+def test_calculate_standardised_anomalies_gridded_data(
+    request, fixture_name, mean_cube, variance_cube
+):
+    """Test that the plugin returns a cube with expected data."""
+    diagnostic_cube = request.getfixturevalue(fixture_name)
+    plugin = CalculateClimateAnomalies()
+    if fixture_name == "diagnostic_cube":
+        expected_anomalies = np.array([3.5], dtype=np.float32).reshape(1, 1, 1)
+    elif fixture_name == "diagnostic_cube_multiple_time_points":
+        expected_anomalies = np.array([1.0, 0.0, -1.0], dtype=np.float32).reshape(
+            3, 1, 1
+        )
+    result = plugin.process(diagnostic_cube, mean_cube, variance_cube)
+    np.testing.assert_allclose(result.data, expected_anomalies, rtol=1e-5)
+    assert result.long_name == diagnostic_cube.name() + "_standard_anomaly"
+    assert result.units == "1"
+    assert "reference_epoch" in [coord.name() for coord in result.coords()]
+    assert (
+        "(CellMethod(method='anomaly', coord_names=('reference_epoch',), intervals=(), comments=())"
+        in str(result.cell_methods)
+    )
+
+
+def test_calculate_unstandardised_anomalies_site_data(site_cubes):
+    """Test that the plugin calculates unstandardised anomalies correctly for site data."""
+    plugin = CalculateClimateAnomalies()
+    site_cube_diagnostic, site_cube_mean, _ = site_cubes
+    result = plugin.process(site_cube_diagnostic, site_cube_mean)
+    expected_anomalies = np.array([7.0], dtype=np.float32)
+    np.testing.assert_allclose(result.data, expected_anomalies, rtol=1e-5)
+    assert result.name() == site_cube_diagnostic.name() + "_anomaly"
+    assert result.units == "K"
+    assert "reference_epoch" in [coord.name() for coord in result.coords()]
+    assert (
+        "(CellMethod(method='anomaly', coord_names=('reference_epoch',), intervals=(), comments=())"
+        in str(result.cell_methods)
+    )
+
+
+def test_calculate_standardised_anomalies_site_data(site_cubes):
+    """Test that the plugin calculates standardised anomalies correctly for site data."""
+    plugin = CalculateClimateAnomalies()
+    site_cube_diagnostic, _, _ = site_cubes
+    result = plugin.process(*site_cubes)
+    expected_anomalies = np.array([3.5], dtype=np.float32)
+    np.testing.assert_allclose(result.data, expected_anomalies, rtol=1e-5)
+    assert result.long_name == site_cube_diagnostic.name() + "_standard_anomaly"
```

Comment:
I think we should also check the `reference_epoch` coordinate is correct, for example containing the correct time and time bounds.